pentasharp/Controllers/BusSchedule.cs

```diff
@@ -1,21 +1,336 @@
-﻿using Microsoft.AspNetCore.Mvc;
-using System.Diagnostics;
-using WebApplication1.Models;
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using pentasharp.Data;
+using pentasharp.Models.Entities;
+using pentasharp.Models.Enums;
+using pentasharp.ViewModel.BusSchedul;
+using pentasharp.ViewModel.BusReservation;
 using WebApplication1.Filters;
+using Microsoft.AspNetCore.Authorization;
 
 namespace WebApplication1.Controllers
 {
-    [ServiceFilter(typeof(AdminOnlyFilter))]
+    [Route("api/BusSchedule")]
     public class BusScheduleController : Controller
     {
+        private readonly AppDbContext _context;
+        private readonly IMapper _mapper;
 
-        public IActionResult BusScheduleManagement()
+        public BusScheduleController(AppDbContext context, IMapper mapper)
         {
-            return View();
+            _context = context;
+            _mapper = mapper;
+        }
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        public IActionResult BusScheduleManagement() => View();
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        [HttpPost("AddRoute")]
+        public IActionResult AddRoute([FromBody] AddRouteViewModel model, int hours, int minutes)
+        {
+            if (!ModelState.IsValid || string.IsNullOrWhiteSpace(model.FromLocation) || string.IsNullOrWhiteSpace(model.ToLocation))
+                return BadRequest(new { success = false, message = "Invalid data provided." });
+
+            if (_context.BusRoutes.Any(r => r.FromLocation == model.FromLocation && r.ToLocation == model.ToLocation))
+                return BadRequest(new { success = false, message = "Route already exists." });
+
+            var route = new BusRoutes
+            {
+                FromLocation = model.FromLocation,
+                ToLocation = model.ToLocation,
+                EstimatedDuration = new TimeSpan(hours, minutes, 0),
+                CreatedAt = DateTime.Now
+            };
+
+            _context.BusRoutes.Add(route);
+            _context.SaveChanges();
+
+            return Ok(new { success = true, message = "Route added successfully." });
+        }
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        [HttpGet("GetRoutes")]
+        public IActionResult GetRoutes()
+        {
+            var routes = _context.BusRoutes.ToList();
+            var routeViewModels = _mapper.Map<List<AddRouteViewModel>>(routes);
+            return Ok(routeViewModels);
+        }
+
+        [ServiceFilter(typeof(AdminOnlyFilter))]
+        [HttpPut("EditRoute/{id}")]
+        public IActionResult EditRoute(int id, [FromBody] AddRouteViewModel model, int hours, int minutes)
+        {
```

Comment:
use LINQ, you have an explanation on why somewhere in the comments i left in this class